orchagent/pfcwdorch.cpp

```diff
@@ -596,13 +592,13 @@ void PfcWdSwOrch<DropHandler, ForwardHandler>::doTask(swss::NotificationConsumer
                         entry->first,
                         entry->second.index,
                         PfcWdOrch<DropHandler, ForwardHandler>::getCountersTable());
+                entry->second.handler->initCounters();
             }
         }
         else
         {
             throw runtime_error("Unknown PFC WD action");
         }
-        entry->second.handler->initCounters();
```

Comment:
What is an actual problem here? If the handler is not created, we throw an exception.

Reply:
If receiving multiple same storm notifications it init counters multiple times

Reply:
Don't you think it is a workaround rather than solution?
What's the reason that we get multiple notifications for same port/queue?

Reply:
I would think this is an allowed behavior for our design. swss is a single thread, and queues all the notification and handle them one by one. If swss cannot handle pfcw storm/restore notification from syncd in time, it's possible that syncd send multiple notifications until swss handles it. This is possible to happen when during swss start time or swss busy updating large amount of routes. Therefore, in the pfcwdorch we should keep this in mind and be robust to handle this kind of case as well. It's not a workaround, unless changing the whole design.

Reply:
ok